lib/autoproj/daemon/git_api/services/gitlab.rb

```diff
@@ -150,12 +150,15 @@ def rate_limit
                         RateLimit.new(1000, 0)
                     end
 
+                    # rubocop: disable Metrics/AbcSize
                     def exception_adapter
```

Comment:
A IMO much better way would be to define the list of connection failed exceptions and use it like this:

~~~
CONNECTION_FAILED_EXCEPTIONS = [....].freeze
def exception_adapter
   yield
rescue Gitlab::Error::NotFound => e
    raise GitAPI::NotFound, e.message
rescue *CONNECTION_FAILED_EXCEPTIONS => e
   raise GitAPI::ConnectionFailed, e.message
end